File: CareConnect/CareConnect.Models/Responses/ReportData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Responses
{
    public class ReportData
    {
        public DateTime Date { get; set; }

        public int NewClients { get; set; }

        public int Appointments { get; set; }

        public int Workshops { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/responses/report_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'report_data.g.dart';

@JsonSerializable()
class ReportData {
  final DateTime date;
  final int newClients;
  final int appointments;
  final int workshops;

  ReportData({
    required this.date,
    required this.newClients,
    required this.appointments,
    required this.workshops,
  });

  factory ReportData.fromJson(Map<String, dynamic> json) =>
      _$ReportDataFromJson(json);

  Map<String, dynamic> toJson() => _$ReportDataToJson(this);
}
